File: solutions/python/robot-name/1/robot_name.py
import random as rd
LETTERS='ABCDEFGHIJKLMNOPQRSTUVWXYZ'
class Robot:
    used_names=[]
    def __init__(self):
        self.name = LETTERS[rd.randint(0, 25)] + LETTERS[rd.randint(0, 25)] + str(rd.randint(0, 9)) + str(rd.randint(0, 9)) + str(rd.randint(0, 9))
        self.used_names.append(self.name)
    def reset(self):
        while True:
            candidate = LETTERS[rd.randint(0, 25)] + LETTERS[rd.randint(0, 25)] + str(rd.randint(0, 9)) + str(rd.randint(0, 9)) + str(rd.randint(0, 9))
            if candidate not in self.used_names:
                self.name=candidate
                break 
            
            
        

File: solutions/javascript/robot-name/1/robot-name.js
export class Robot {
  static releaseNames = () => Robot.usedNames = new Set()
  static usedNames = new Set()
  static generateName = () => {
    const alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    const randomLetter = () => alphabet[Math.floor(Math.random() * 26)]
    return `${randomLetter()}${randomLetter()}${Math.floor(Math.random() * 9)}${Math.floor(Math.random() * 9)}${Math.floor(Math.random() * 9)}`
  };
  get name() {
    return this._name
  }
  constructor() {
    let name = Robot.generateName();
    while (Robot.usedNames.has(name)){
      name = Robot.generateName();
    }
    this._name = name;
    Robot.usedNames.add(name)
  }
  reset() {
    let name = Robot.generateName();
    while (true){
      if (Robot.usedNames.has(name)){
        name = Robot.generateName();
      } else {
        break;
      }
    }
    this._name = name;
    Robot.usedNames.add(name);
  }
}